framework/aster-frame/src/mm/space.rs

```diff
@@ -202,12 +234,57 @@ impl VmSpace {
     /// read-only. And both the VM space will take handles to the same
     /// physical memory pages.
     pub fn fork_copy_on_write(&self) -> Self {
+        let page_fault_handler = Once::new();
+        if let Some(handler) = self.page_fault_handler.get() {
+            page_fault_handler.call_once(|| *handler);
+        }
         let new_space = Self {
             pt: self.pt.fork_copy_on_write(),
+            page_fault_handler,
         };
         tlb_flush_all_excluding_global();
         new_space
     }
+
+    /// Returns a reader to read data from the user space of the current task.
+    pub fn current_reader(vaddr: Vaddr, len: usize) -> Result<VmReader<'static, UserSpace>> {
+        let current_task = current_task().ok_or(Error::AccessDenied)?;
+        let user_space = current_task.user_space().ok_or(Error::AccessDenied)?;
```

Comment:
Seems that they are not triggered by a permission check. Maybe create an error type for it. Or leave the return value as `Option<VmReader<'static, UserSpace>>`.